src/libxively/sft/xi_sft_logic.c

```diff
@@ -268,6 +270,9 @@ xi_sft_on_message( xi_sft_context_t* context, xi_control_message_t* sft_message_
                             XI_CONTROL_MESSAGE__SFT_FILE_STATUS_CODE_SUCCESS );
                     }
 
+                    xi_sft_revision_set( context->update_current_file->name,
```

Comment:
hm, good question, haven't thought about this yet. And maybe checksum should happen even before writing the file itself with final filename onto the flash storage. Since a broken content may overwrite the previous old but intact file.